src/more_ops.rs

```diff
@@ -532,18 +485,21 @@ pub fn op_ash<T: Allocator>(a: &mut T, input: T::Ptr, _max_cost: Cost) -> Respon
 
 pub fn op_lsh<T: Allocator>(a: &mut T, input: T::Ptr, _max_cost: Cost) -> Response<T::Ptr> {
     let args = Node::new(a, input);
-    let (i0, l0, i1, _) = uint_int(&args, "lsh")?;
-    let s1 = i64::try_from(&i1);
-    if match s1 {
-        Err(_) => true,
-        Ok(v) => v.abs() > 65535,
-    } {
+    check_arg_count(&args, 2, "lsh")?;
+    let a0 = args.first()?;
+    let b0 = int_atom(&a0, "lsh")?;
+    let i0 = BigUint::from_bytes_be(b0);
```

Comment:
This is kind of an annoyance, as I think it's the only place in an operator impl that exposes which bigint library we're using underneath. Moving into `Number` might make gmp port less of a headache. But I guess we can burn that bridge when we get to it. [sic]

Reply:
yes. In fact, since a shift is a pretty simple operation, we might even get away with implementing it ourselves, over the byte-buffer. But I think we can punt on that